mongorestore/mongorestore_test.go

```diff
@@ -2345,3 +2347,228 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
 func uniqueDBName() string {
 	return fmt.Sprintf("mongorestore_test_%d_%d", os.Getpid(), time.Now().UnixMilli())
 }
+
+// TestRestoreColumnstoreIndex tests restoring Columnstore Indexes in restored collections.
+func TestRestoreColumnstoreIndex(t *testing.T) {
+	require := require.New(t)
+
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	require.NoError(err, "can connect to server")
+
+	fcv := testutil.GetFCV(session)
+	if cmp, err := testutil.CompareFCV(fcv, "6.3"); err != nil || cmp < 0 {
+		t.Skipf("Requires server with FCV 6.3 or later and we have %s", fcv)
+	}
+
+	t.Run("restore from dump", func(t *testing.T) {
+		testRestoreColumnstoreIndexFromDump(t)
+	})
+
+	res := session.Database("admin").RunCommand(context.Background(), bson.M{"replSetGetStatus": 1})
+	if res.Err() != nil {
+		t.Skip("server is not part of a replicaset so we cannot test restore from oplog")
+	}
+	t.Run("restore from oplog", func(t *testing.T) {
+		testRestoreColumnstoreIndexFromOplog(t)
+	})
+}
+
+// testRestoreColumnstoreIndexFromDump tests restoring Columnstore Indexes from dump files.
+func testRestoreColumnstoreIndexFromDump(t *testing.T) {
+	require := require.New(t)
+
+	session, err := testutil.GetBareSession()
+	require.NoError(err, "can connect to server")
+
+	dbName := uniqueDBName()
+	testDB := session.Database(dbName)
+	defer func() {
+		err = testDB.Drop(nil)
+		if err != nil {
+			t.Fatalf("Failed to drop test database: %v", err)
+		}
+	}()
+
+	key := "$**"
+	columnstoreProjection := map[string]int32{"price": 0}
```

Comment:
While `: 0` apparently works, all the doc examples I've seen for indexes use `: 1` so let's use that instead.